src/util.c

```diff
@@ -431,6 +434,142 @@ int write_result_into_log_file(struct ntttcp_test_endpoint *tep)
+	unsigned int i;
+
+	memset(str_temp1, '\0', sizeof(char)*256);
+	memset(str_temp2, '\0', sizeof(char)*2048);
+
+	FILE *json_file = fopen(test->json_log_filename, "w");
+	if (json_file == NULL) {
+		PRINT_ERR("Error opening file to write log");
+		return -1;
+	}
+
+	gethostname(str_temp1, 256);
+	fprintf(json_file, "{\n");
+	fprintf(json_file, " \"ntttcp%s\" : {\n", tep->endpoint_role == ROLE_RECEIVER ? "r" : "s");
+	fprintf(json_file, "  \"-computername\" : \"%s\",\n", str_temp1);
+	fprintf(json_file, "  \"-version\" : \"5.33-linux\",\n");
+	fprintf(json_file, "  \"parameters\" : {\n");
+	fprintf(json_file, "   \"sendSocketBuff\" : \"%lu\",\n", test->send_buf_size);
+	fprintf(json_file, "   \"recvSocketBuff\" : \"%lu\",\n", test->recv_buf_size);
+	fprintf(json_file, "   \"port\" : \"%d\",\n", test->server_base_port);
+	fprintf(json_file, "   \"syncPort\" : \"%d\",\n", test->server_base_port - 1);
+	fprintf(json_file, "   \"noSync\" : \"%s\",\n", test->no_synch == 0 ? "False" : "True");
+	fprintf(json_file, "   \"waitTimeoutMilliseconds\" : \"%d\",\n", 0);
+	fprintf(json_file, "   \"async\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"verbose\" : \"%s\",\n", test->verbose ? "True" : "False");
+	fprintf(json_file, "   \"wsa\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"useIpv6\" : \"%s\",\n", test->domain == AF_INET6 ? "True" : "False");
+	fprintf(json_file, "   \"udp\" : \"%s\",\n", test->protocol == UDP ? "True" : "False");
+	fprintf(json_file, "   \"verifyData\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"waitAll\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"runTime\" : \"%d\",\n", test->duration);
+	fprintf(json_file, "   \"warmupTime\" : \"%d\",\n", test->warmup);
+	fprintf(json_file, "   \"cooldownTime\" : \"%d\",\n", test->cooldown);
+	fprintf(json_file, "   \"dashNTimeout\" : \"%d\",\n", 0);
+	fprintf(json_file, "   \"bindSender\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"senderName\" : \"%s\",\n", "NA");
+	fprintf(json_file, "   \"maxActiveThreads\" : \"%d\",\n", 0);
+	fprintf(json_file, "   \"tp\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"noStdioBuffer\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"throughputBpms\" : \"%d\",\n", 0);
+	fprintf(json_file, "   \"cpuBurn\" : \"%d\",\n", 0);
+	fprintf(json_file, "   \"latencyMwasurement\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"useIOComplPorts\" : \"%s\",\n", "NA");
+	fprintf(json_file, "   \"cpuFromIdleFlag\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"getEstats\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"qosFlag\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"jitterMeasurement\" : \"%s\",\n", "False");
+	fprintf(json_file, "   \"packetSpacing\" : \"%d\"\n", 0);
+	fprintf(json_file, "  },\n");
+
+	fprintf(json_file, "  \"threads\" : [\n");
+	for(i = 0; i < tep->total_threads; i++ ) {
+		if (tep->results->threads[i]->is_sync_thread == true && i == tep->total_threads - 1) { 
+			fprintf(json_file, "   }\n");
```

Comment:
If the rule that the sync thread is the last thread won't be changed, we can delete other condition judgment, just keep one "if (tep->results->threads[i]->is_sync_thread == true)", and we need to add a comment to explain this situation. 

Otherwise, we can't assume sync thread is the last one. If the sync thread is changed into the first one, we should make sure the json log also has the right format. If so, we can write like this:
```
        if (tep->results->threads[i]->is_sync_thread == true) {
            if (i == tep->total_threads - 1)
                fprintf(json_file, "   }\n");
            else if (i != 0)
                fprintf(json_file, "  },\n");
            continue;
        } else if (i != 0)
            fprintf(json_file, "  },\n");
```

Reply:
done